package.json

```diff
@@ -24,8 +24,9 @@
     "prettier": "prettier --write ."
   },
   "dependencies": {
-    "@nivo/core": "^0.74.0",
-    "@nivo/line": "^0.74.0",
+    "@nivo/core": "^0.78.0",
+    "@nivo/line": "^0.78.0",
+    "@nivo/pie": "^0.78.0",
```

Comment:
As already mentioned, I am concerned with increasing the package lock size that much. With this changeset, there is more than twice as many dependency lines in this website as there is in the core, and all of that for a simple static website.

If the only reason for this massive increase is the `pie` dependency, I would prefer that we include the pie graph as a static image, or even just as a table.

Reply:
Concerning the size of the javascript loaded, I can make sure to load the  dependencies only on the page that uses it with the [dynamic import](https://nextjs.org/docs/advanced-features/dynamic-import). It is not an inexpensive improvement and @martinratinaud can help me.

Concerning the numbers of the dependency for the simple static website, i am not in favour of using an image and a table because:
- will be more complicated to update 
- would probably make the page much less pretty 
- I find it hard to let go of the idea that much of the work I have done on this PR is being thrown away

Reply:
The resulting homepage goes from 1.48MB to 1.69MB (+14%) for a page that very few users will see. I personally do find it regrettable, but lacking a clear agreement on ecological responsibility and accessibility of our work, I will not block further against the feeling of [sunk costs](https://thedecisionlab.com/biases/the-sunk-cost-fallacy).

Reply:
You're right @MattiSG and there is a tool for that in nextjs called `dynamic`.
Using this feature will enable chunking in webpack, leading to reduce package size.

I've done so in [this commit](https://github.com/ambanum/OpenTermsArchive.org/pull/154/commits/883f44aa746c9ec04678c300c09e6c4606326e08)

Building the website with `npm run build` gives a budget page size of 149 kB.

Reply:
Thanks @martinratinaud!
Unfortunately following a force-push the commit reference is broken, but I assume it is https://github.com/ambanum/OpenTermsArchive.org/pull/154/commits/7b2e33a75be728fc2e1cf9c81b4c582b1e3d4cfe 🙂

Since the files have been modified significantly after that, commit, I am curious how you measured the 149kB figure. With the Safari inspector, I now get a ~1.7MB figure on the homepage after scrolling down.

Reply:
Below the result of `npm run build`


```console
Route (pages)                              Size     First Load JS
┌ ● /                                      266 B           154 kB
├   /_app                                  0 B             120 kB
├ ○ /404                                   195 B           120 kB
├ ● /about                                 5.08 kB         150 kB
├   └ css/b3234b3a39c46e16.css             3.02 kB
├ ● /accessibility                         2.25 kB         147 kB
├ λ /api/ota/contributors/all              0 B             120 kB
├ λ /api/ota/expenses                      0 B             120 kB
├ λ /api/ota/services/all                  0 B             120 kB
├ λ /api/ota/services/graph                0 B             120 kB
├ λ /api/ota/stats/contributors            0 B             120 kB
├ λ /api/ota/versions/commits/latest       0 B             120 kB
├ λ /api/subscribe                         0 B             120 kB
├ ● /budget                                4.01 kB         149 kB
├   └ css/32ac23686a97c76e.css             2.52 kB
├ ● /case-studies                          2.69 kB         148 kB
├   └ css/6e6fb6529c008f40.css             2.74 kB
├ ● /design                                7.64 kB         157 kB
├   └ css/44cbdc7d1c89941e.css             2.36 kB
├ ● /homepage                              214 B           154 kB
├ ● /legal-notice                          2.26 kB         148 kB
├ ● /media                                 4.16 kB         149 kB
├   └ css/1c75cd34940cee85.css             1.2 kB
├ ● /privacy-policy                        2.26 kB         148 kB
├ ● /stats                                 4.29 kB         154 kB
├   └ css/b51c4581ceed2c30.css             3.45 kB
├ ● /subscribe                             12.2 kB         157 kB
├   └ css/e6ace283c34daf4f.css             182 B
├ ● /subscribe/confirm                     2.26 kB         148 kB
├ ● /subscribe/done                        2.25 kB         147 kB
└ ● /subscribe/french-elections            2.27 kB         148 kB
+ First Load JS shared by all              125 kB
  ├ chunks/framework-bb5c596eafb42b22.js   42.1 kB
  ├ chunks/main-58e03d197406de17.js        35.6 kB
  ├ chunks/pages/_app-1bdd2b024ae1cb17.js  40.5 kB
  ├ chunks/webpack-c140a1a7eda20828.js     1.89 kB
  └ css/a8ee84aaaccdd6a7.css               4.82 kB

ƒ Middleware                               24 kB

λ  (Server)  server-side renders at runtime (uses getInitialProps or getServerSideProps)
○  (Static)  automatically rendered as static HTML (uses no initial props)
●  (SSG)     automatically generated as static HTML + JSON (uses getStaticProps)
```